bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/coap/ShellyCoapHandler.java

```diff
@@ -698,48 +822,60 @@ private void resetSerial() {
      * @param sensorId The id from the sensor update
      * @return Index of found entry (+1 will be the suffix for the channel name) or null if sensorId is not found
      */
-    @SuppressWarnings("null")
-    @Nullable
-    private Integer getInputId(String sensorId) {
-        Integer idx = 0;
+    private int getSensorNumber(String sensorName, String sensorId) {
+        int idx = 0;
         for (Map.Entry<String, CoIotDescrSen> se : sensorMap.entrySet()) {
-            @Nullable
             CoIotDescrSen sen = se.getValue();
-            if (sen.id.equalsIgnoreCase("Input")) {
+            if (sen.desc.equalsIgnoreCase(sensorName)) {
                 idx++; // iterate from input1..2..n
             }
-            if (sen.id.equalsIgnoreCase(sensorId)) {
+            if (sen.id.equalsIgnoreCase(sensorId) && blockMap.containsKey(sen.links)) {
                 CoIotDescrBlk blk = blockMap.get(sen.links);
-                if ((blk != null) && StringUtils.substring(blk.desc, 5).equalsIgnoreCase("Relay")) {
+                if (StringUtils.substring(blk.desc, 5).equalsIgnoreCase("Relay")) {
                     idx = Integer.parseInt(StringUtils.substringAfter(blk.desc, "Relay"));
                 }
-                logger.trace("{}:    map to input{} channel", thingName, idx);
+                logger.trace("{}:    map sensor {}{} to index {}", thingName, sensorName, sensorId, idx);
+                return idx;
+            }
+        }
+        logger.debug("{}: sensorId {} not found in sensorMap!", thingName, sensorId);
+        return -1;
+    }
+
+    private int getExtTempId(String sensorId) {
+        int idx = 0;
+        for (Map.Entry<String, CoIotDescrSen> se : sensorMap.entrySet()) {
```

Comment:
I assume that sensorMap is a LinkedHashMap or an immutable map otherwise this id generation can get very inconsistent.

Reply:
changed to LinkedHashMap